src/utils.rs

```diff
@@ -1,12 +1,46 @@
 extern crate base64;
 
 use self::base64::encode;
+use http::retrieve_asset;
 use regex::Regex;
+use std::collections::HashMap;
 use url::{ParseError, Url};
 
+/// This monster of a regex is used to match any kind of URL found in a CSS
+/// stylesheet.
+///
+/// There's roughly three different categories that a found url could fit into:
+///    - Font       [found after a src: property in an @font-family rule]
+///    - Stylesheet [denoted by an @import before the url
+///    - Image      [covers all other uses of the url() function]
+///
+/// This regex aims to extract the following information:
+///    - What type of URL is it (font/image/css)
+///    - Where is the part that needs to be replaced (incl any wrapping quotes)
+///    - What is the URL (excl any wrapping quotes)
+///
+/// For understandablility, the regex can be broken down into two parts:
+///
+/// `(?:(?P<import>@import)|(?P<font>src\s*:)\s+)?`
+/// This matches the precursor to a font or css url, and fills in a match under
+/// either `<import>` (if it's a css url) or `<font>` (if it's a font).
+/// Determining whether or not it's an image can be done by the negation of both
+/// of these.  Either zero or one of these can match.
```

Comment:
There seems to be two spaces after "these."

Reply:
When I was growing up, that's how we were taught to type spaces after periods, interestingly, but if that's not what's going on in the rest of the documentation, I have no problem removing it.

Reply:
Interesting, didn't know that.
> If you learned to type on a typewriter before word processors became the norm, two spaces after a period were required and taught as correct. The extra space was needed to delineate the beginning of a new sentence because the spacing between words was uneven on a typewriter.